ctapipe/reco/weighted_axis_minimisation.py

```diff
@@ -0,0 +1,296 @@
+
+    @staticmethod
+    def rotate_translate(pixel_pos_x,pixel_pos_y,x_trans,y_trans,phi):
+        """
+        Function to perform rotation and translation of pixel lists
+
+        Parameters
+        ----------
+        pixel_pos_x: ndarray
+            Array of pixel x positions
+        pixel_pos_y: ndarray
+            Array of pixel x positions
+        x_trans: float
+            Translation of position in x coordinates
+        y_trans: float
+            Translation of position in y coordinates
+        phi: float
+            Rotation angle of pixels
+
+        Returns
+        -------
+            ndarray,ndarray: Transformed pixel x and y coordinates
+        """
+        pixel_pos_trans_x = (pixel_pos_x-x_trans) *np.cos(phi) - (pixel_pos_y-y_trans)*np.sin(phi)
+        pixel_pos_trans_y = (pixel_pos_x-x_trans) *np.sin(phi) + (pixel_pos_y-y_trans)*np.cos(phi)
+
+        return pixel_pos_trans_x,pixel_pos_trans_y
+
+    def get_dist_from_axis(self,pixel_pos_x,pixel_pos_y,x_trans,y_trans,phi,pixel_weight):
+        """
+        Function to perform weighting of pixel positions from the shower axis for an individual image
+
+        Parameters
+        ----------
+        pixel_pos_x: ndarray
+            Array of pixel x positions
+        pixel_pos_y: ndarray
+            Array of pixel x positions
+        x_trans: float
+            Translation of position in x coordinates
+        y_trans: float
+            Translation of position in y coordinates
+        phi: float
+            Rotation angle of pixels
+        pixel_weight: ndarray
+            Weighting of individual pixels (usually amplitude)
+
+        Returns
+        -------
+            Weighted sum of distances of pixels from shower axis
+        """
+        pixel_pos_x_trans,pixel_pos_y_trans = self.rotate_translate(pixel_pos_x,pixel_pos_y,x_trans,y_trans,phi)
+        pixel_pos_y_trans *= pixel_weight
+        return np.sum(np.abs(pixel_pos_y_trans))
+
+    def weighted_dist(self,x_src,y_src,x_grd,y_grd):
+        """
+        Function to be minimised to find core and source position.
+        Calculates expected image axis in nominal system and returns the weighted sum
+        of distances from the predicted image axis.
+
+        Parameters
+        ----------
+        x_src: float
+            Test source position in nominal system
+        y_src: float
+            Test source position in nominal system
+        x_grd: float
+            Test core position in tilted system
+        y_grd: float
+            Test core position in tilted system
+
+        Returns
+        -------
+            Sum of weighted pixel distances from predicted axis
+        """
+
+        sum = 0
+        for pos_x,pos_y,tel_x,tel_y,weight in zip(self.pixel_pos_x,self.pixel_pos_y,self.tel_pos_x,self.tel_pos_y,self.pixel_weight):
+            phi = np.arctan2((tel_y-y_grd),(tel_x-x_grd)) * u.rad
+            sum += self.get_dist_from_axis(pos_x,pos_y,x_src,y_src,phi,weight)
```

Comment:
As far as I see it, this complete loop could be replaced by a single `np.sum` if `pixel_pos_*` were a 2d array and not a list.